tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -79,19 +95,65 @@ public void Run(JsonDrivenTestCase testCase)
         }
 
         // private methods
-        private void Run(BsonDocument shared, BsonDocument test)
+        private void CheckClusterTopology(BsonArray allowedTopologies)
+        {
+            var allowedClusterTypes = allowedTopologies.Select(MapToplogyToClusterType);
+            RequireServer.Check().ClusterTypes(allowedClusterTypes.ToArray());
+
+            var clusterType = CoreTestConfiguration.Cluster.Description.Type;
+            switch (clusterType)
+            {
+                case ClusterType.Sharded:
+                    RequireServer.Check().Supports(Feature.ShardedTransactions);
+                    break;
+                case ClusterType.ReplicaSet:
+                    RequireServer.Check().Supports(Feature.Transactions);
+                    break;
+                default:
+                    throw new Exception("Topology type for transactions must be replicaset or sharded.");
+
+            }
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!disposing) return;
+            foreach (var disposable in _disposables)
+            {
+                disposable.Dispose();
+            }
+        }
+
+        private ClusterType MapToplogyToClusterType(BsonValue topology)
         {
-            if (test.Contains("skipReason"))
+            switch (topology.AsString)
             {
-                throw new SkipException(test["skipReason"].AsString);
+                case "single": return ClusterType.Standalone;
+                case "replicaset": return ClusterType.ReplicaSet;
+                case "sharded": return ClusterType.Sharded;
+                default: throw new Exception("Unknown topology type");
             }
-            //if (test["description"].AsString != "rerun commit after empty transaction")
-            //{
-            //    return;
-            //}
+        }
 
-            JsonDrivenHelper.EnsureAllFieldsAreValid(shared, "_path", "database_name", "collection_name", "data", "tests");
-            JsonDrivenHelper.EnsureAllFieldsAreValid(test, "description", "clientOptions", "failPoint", "sessionOptions", "operations", "expectations", "outcome", "async");
+        private void Run(BsonDocument shared, BsonDocument test)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(shared,
+                "_path",
+                "database_name",
+                "collection_name",
+                "data",
+                "tests",
+                "topology");
+            JsonDrivenHelper.EnsureAllFieldsAreValid(test,
+                "description",
+                "clientOptions",
+                "failPoint",
+                "sessionOptions",
+                "operations",
+                "expectations",
+                "outcome",
+                "async",
+                "useMultipleMongoses");
```

Comment:
This is where I would check for skipReason and topology:

            if (test.Contains("skipReason"))
            {
                throw new SkipException($"Test skipped because {test["skipReason"].AsString}.");
            }

            if (shared.TryGetValue("topology", out var topology))
            {
                CheckClusterTopology(topology.AsBsonArray);
            }

Note: I think it makes sense to check for skipReason first. We don't want to throw a topology skip message when the test was supposed to be skipped in the first place.